mirgecom/boundary.py

```diff
@@ -666,7 +882,1095 @@ def isothermal_noslip_state(
+                                       grad_cv_minus=grad_cv_minus,
+                                       normal=normal, **kwargs)
+
+        grad_t_plus = self._bnd_grad_temperature_func(
+            dcoll=dcoll, dd_bdry=dd_bdry, gas_model=gas_model,
+            state_minus=state_minus, grad_cv_minus=grad_cv_minus,
+            grad_t_minus=grad_t_minus)
+
+        # Note that [Mengaldo_2014]_ uses F_v(Q_bc, dQ_bc) here and
+        # *not* the numerical viscous flux as advised by [Bassi_1997]_.
+        f_ext = viscous_flux(state=state_plus, grad_cv=grad_cv_plus,
+                             grad_t=grad_t_plus)
+        return f_ext@normal
+
+
+class AdiabaticNoslipWallBoundary(PrescribedFluidBoundary):
+    r"""Adiabatic viscous wall boundary.
+
+    This class implements an adiabatic no-slip wall consistent with the prescription
+    by [Mengaldo_2014]_.
+
+    .. automethod:: inviscid_wall_flux
+    .. automethod:: viscous_wall_flux
+    .. automethod:: grad_cv_bc
+    .. automethod:: temperature_bc
+    .. automethod:: adiabatic_wall_state_for_advection
+    .. automethod:: adiabatic_wall_state_for_diffusion
+    .. automethod:: grad_temperature_bc
+    .. automethod:: adiabatic_noslip_grad_av
+    """
+
+    def __init__(self):
+        """Initialize the boundary condition object."""
+        PrescribedFluidBoundary.__init__(
+            self, boundary_state_func=self.adiabatic_wall_state_for_diffusion,
+            inviscid_flux_func=self.inviscid_wall_flux,
+            viscous_flux_func=self.viscous_wall_flux,
+            boundary_temperature_func=self.temperature_bc,
+            boundary_gradient_cv_func=self.grad_cv_bc
+        )
+
+    def adiabatic_wall_state_for_advection(self, dcoll, dd_bdry, gas_model,
+                                           state_minus, **kwargs):
+        """Return state with zero-velocity."""
+        dd_bdry = as_dofdesc(dd_bdry)
+        mom_plus = -state_minus.momentum_density
+        cv_plus = make_conserved(
+            state_minus.dim, mass=state_minus.mass_density,
+            energy=state_minus.energy_density, momentum=mom_plus,
+            species_mass=state_minus.species_mass_density
+        )
+        return make_fluid_state(cv=cv_plus, gas_model=gas_model,
+                                temperature_seed=state_minus.temperature)
+
+    def adiabatic_wall_state_for_diffusion(self, dcoll, dd_bdry, gas_model,
+                                           state_minus, **kwargs):
+        """Return state with zero-velocity."""
+        dd_bdry = as_dofdesc(dd_bdry)
+        mom_plus = 0.*state_minus.momentum_density
+        cv_plus = make_conserved(
+            state_minus.dim, mass=state_minus.mass_density,
+            energy=state_minus.energy_density, momentum=mom_plus,
+            species_mass=state_minus.species_mass_density
+        )
+        return make_fluid_state(cv=cv_plus, gas_model=gas_model,
+                                temperature_seed=state_minus.temperature)
```

Comment:
Should we be modifying the (kinetic) energy here, like we do for the isothermal no-slip? (If not, maybe add a comment explaining why for clarity.)

Reply:
Yes, I think this is fixed now.